packages/core/src/components/forms/numericInput.tsx

```diff
@@ -0,0 +1,423 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+
+import {
+    AbstractComponent,
+    Classes,
+    HTMLInputProps,
+    IIntentProps,
+    IProps,
+    Keys,
+    Position,
+    removeNonHTMLProps,
+    Utils,
+} from "../../common";
+import * as Errors from "../../common/errors";
+
+import { Button } from "../button/buttons";
+import { InputGroup } from "./inputGroup";
+
+export interface INumericInputProps extends IIntentProps,
+IProps {
+
+    /**
+     * The position of the buttons with respect to the input field.
+     * @default Position.RIGHT
+     */
+    buttonPosition?: Position.LEFT | Position.RIGHT | "none";
+
+    /**
+     * Whether the input is in a non-interactive state.
+     * @default false
+     */
+    disabled?: boolean;
+
+    /** The name of icon (the part after `pt-icon-`) to render on left side of input. */
+    leftIconName?: string;
+
+    /** The placeholder text in the absence of any value. */
+    placeholder?: string;
+
+    /**
+     * The increment between successive values when `shift` is held.
+     * @default 10
+     */
+    majorStepSize?: number;
+
+    /** The maximum value of the input. */
+    max?: number;
+
+    /** The minimum value of the input. */
+    min?: number;
+
+    /**
+     * The increment between successive values when `alt` is held.
+     * @default 0.1
+     */
+    minorStepSize?: number;
+
+    /**
+     * The increment between successive values when no modifier keys are held.
+     * @default 1
+     */
+    stepSize?: number;
+
+    /** The value to display in the input field. */
+    value?: number | string;
+
+    /** The callback invoked when `enter` is pressed and when the field loses focus. */
+    onConfirm?(value: number | string): void;
+
+    /** The callback invoked when the value changes. */
+    onChange?(value: number | string): void;
```

Comment:
how about `onNumberChange(value: number)` which only gets invoked when the value changes and is a valid number? if users want more fine-grained string change then they should be able to use HTML `onChange` directly and do their own event management.